### PR TITLE
documentation #816: updated swagger for permission and telegram group…

### LINKED ISSUE #816: Update Swagger documentation for TelegramGroup and Permission
We are updating the documentation for all endpoints and adding more _description and details_. 

You should update the documentation of permission and telegram group according to the **pattern** below:
```ts
  @ApiBearerAuth()
  @ApiOkResponse()
  @ApiBadRequestResponse({
    description: `\n
    InvalidEntityIdException:
      User with such id is not found

    AlreadyRegisteredException:
      User is already registered

    InvalidBodyException:
      Group id can not be empty
      IsCaptain must be a boolean
      IsCaptain can not be empty
    
    AbsenceOfCaptainException:
      Captain was not found
    
    AbsenceOfCaptainTelegramException:
      Captain's telegramId was not found`,
  })
  @ApiUnauthorizedResponse({
    description: `\n
    UnauthorizedException:
      Unauthorized`,
  })
  @ApiForbiddenResponse({
    description: `\n
    NoPermissionException:
      You do not have permission to perform this action
    
    ForbiddenException:
      Forbidden`,
  })
  @ApiParam({
    name: 'userId',
    required: true,
    description: 'Id of a user to request to the group',
  })
  @ApiEndpoint({
    summary: 'Request captain or admin to join the group',
    permissions: PERMISSION.USERS_$USERID_GROUP_REQUEST,
  })
  @Patch('/:userId/requestNewGroup')
  requestNewGroup (
    @Param('userId', UserByIdPipe) userId: string,
    @Body() body: GroupRequestDTO,
  ) {
    return this.userService.requestNewGroup(userId, body);
  }
```
**Structure:**
- auth (if needed)
- ok response
- client errors
- params/query description
- api endpoint for summary, permissions, guards

You should also add **field description** in the body or response.
Add **summary** to endpoints (fix this issue).
Set **default values** for the optional fields

Image
Image

[Our Swagger](https://apidev.fictadvisor.com/api)
[Doc](https://docs.nestjs.com/openapi/introduction)

### PR DESCRIPTION
**Pull request by illiakroshka**
…s endpoints

- POST /v2/permissions/check
- POST /v2/telegramGroups/{groupId}
- GET /v2/telegramGroups/{groupId}
- PATCH /v2/telegramGroups
- DELETE /v2/telegramGroups
- GET /v2/telegramGroups/telegram/{telegramId} 

closes #816

### CODE REVIEW

File: src/v2/api/controllers/PermissionController.ts [hunk 1]
```diff
@@ -35,6 +31,14 @@ export class PermissionController {
       obj.permissions: permissions must be an array
       obj.permissions: each value in permissions must be one of the following values`,
   })
+  @ApiQuery({
+    name: 'userId',
+    required: true,
+  })
```
Comment (JessFreak) [inline src/v2/api/controllers/PermissionController.ts:37]: add description

File: src/v2/api/controllers/TelegramGroupController.ts [hunk 2]
```diff
@@ -180,6 +217,11 @@ export class TelegramGroupController {
   @ApiParam({
     name: 'telegramId',
     type: Number,
+    required: true,
+    description: 'Id of the telegram group to get information',
+  })
+  @ApiEndpoint({
+    summary: 'Get information about telegram group',
   })
```
Comment (JessFreak) [inline src/v2/api/controllers/TelegramGroupController.ts:225]: this endpoint returns information for several telegramGroups, not just one

**Review by JessFreak — CHANGES_REQUESTED** [review]

**Review by JessFreak — APPROVED** [review]

**Review by aim4ik11 — CHANGES_REQUESTED** [review]
Write documentation and add descriptions to CheckPermissionsDTO and CheckPermissionsResponse
image

**Review by aim4ik11 — CHANGES_REQUESTED** [review]
Write documentation and add descriptions to CreateTelegramGroupDTO
image

**Review by aim4ik11 — CHANGES_REQUESTED** [review]
Write documentation and add descriptions to TelegramGroupResponse
image

**Review by aim4ik11 — CHANGES_REQUESTED** [review]
Write documentation and add descriptions to TelegramGroupsResponse
image

**Review by aim4ik11 — CHANGES_REQUESTED** [review]
Write documentation and add descriptions to UpdateTelegramGroupDTO
image

**Review by aim4ik11 — CHANGES_REQUESTED** [review]
Write documentation and add descriptions to TelegramGroupsByTelegramIdResponse, TelegramGroupByTelegramIdResponse, GroupResponse
image

**Review by aim4ik11 — CHANGES_REQUESTED** [review]
Response body in permissions/check does not match with example value response
image

File: src/v2/api/responses/TelegramGroupByTelegramIdResponse.ts [hunk 3]
```diff
@@ -13,10 +14,14 @@ export class TelegramGroupByTelegramIdResponse {
   })
```
Comment (aim4ik11) [inline src/v2/api/responses/TelegramGroupByTelegramIdResponse.ts:14]: Add description to response

**Review by aim4ik11 — CHANGES_REQUESTED** [review]
Fix indent near DataNotFoundException: in update group

image

**Review by aim4ik11 — CHANGES_REQUESTED** [review]
Change decorators order so that first goes auth, created response and then exception responses

image

**Review by aim4ik11 — CHANGES_REQUESTED** [review]
Set correct order in all endpoint in telegramgroupcontroller

File: src/v2/api/controllers/TelegramGroupController.ts [hunk 4]
```diff
@@ -164,8 +184,14 @@ export class TelegramGroupController {
     DataNotFoundException:
       Data were not found`,
   })
+  @ApiParam({
+    name: 'groupId',
+    required: true,
+    description: 'Id of university group from which to get telegram groups',
+  })
+  @UseGuards(TelegramGuard)
```
Comment (aim4ik11) [inline src/v2/api/controllers/TelegramGroupController.ts]: Remove use guards, api endpoint guards is enough

File: src/v2/api/controllers/TelegramGroupController.ts [hunk 5]
```diff
@@ -60,6 +60,12 @@ export class TelegramGroupController {
     InvalidEntityIdException:
       Group with such id is not found`,
   })
+  @ApiParam({
+    name: 'groupId',
+    required: true,
+    description: 'University group id to which to add telegram group',
+  })
+  @UseGuards(TelegramGuard)
```
Comment (aim4ik11) [inline src/v2/api/controllers/TelegramGroupController.ts]: Remove useguards

File: src/v2/api/controllers/TelegramGroupController.ts [hunk 6]
```diff
@@ -116,11 +127,6 @@ export class TelegramGroupController {
   }
 
   @ApiBearerAuth()
-  @ApiUnauthorizedResponse({
-    description: `\n
-    UnauthorizedException:
-      Unauthorized`,
-  })
   @ApiOkResponse()
```
Comment (fokaaas) [inline src/v2/api/controllers/TelegramGroupController.ts]: The deleted group should be here

**Review by fokaaas — CHANGES_REQUESTED** [review]

**Review by aim4ik11 — APPROVED** [review]

**Review by fokaaas — APPROVED** [review]

**Pull request merged**